integ/src/monitor.rs

```diff
@@ -90,23 +90,30 @@ impl<T: BrupopClient> BrupopMonitor<T> {
 
     // verify if Burpop pods (agent, api-server, controller) are in `running` status.
     fn check_pods_health(&self, pods: &ObjectList<Pod>) -> bool {
-        pods.iter().all(|pod| is_pod_running(pod))
+        if pods.items.is_empty() {
+            return false;
+        } else {
+            return pods.iter().all(|pod| is_pod_running(pod));
+        }
     }
 
     // verify if brs has been created properly and initialized `status`.
     async fn check_shadows_health(
         &self,
         bottlerocketshadows: &ObjectList<BottlerocketShadow>,
     ) -> Result<bool> {
```

Comment:
I think we don't need to return a `Result` here, since it seems like the function can't fail. If that's the case, you might be able to simplify this using the iterator pattern you used above.